src/Host/Client/Impl/Session/RSessionProvider.cs

```diff
@@ -35,9 +39,10 @@ public class RSessionProvider : IRSessionProvider {
         public event EventHandler BrokerChanged;
         public event EventHandler<BrokerStateChangedEventArgs> BrokerStateChanged;
 
-        public RSessionProvider(IRSessionProviderCallback callback = null) {
+        public RSessionProvider(IActionLog log, IRSessionProviderCallback callback = null) {
```

Comment:
Can we create a `NullActionLog` and make it a default value for `RSessionProvider`?